core/src/test/java/org/apache/iceberg/TestTransaction.java

```diff
@@ -699,4 +701,38 @@ public void testTransactionRewriteManifestsAppendedDirectly() throws IOException
 
     Assert.assertFalse("Append manifest should be deleted on expiry", new File(newManifest.path()).exists());
   }
+
+  @Test
+  public void testTransactionNotDeletingMetadataOnUnknownSate() throws IOException {
```

Comment:
Should we add tests for the other changed code paths? I notice we handle this in Create,Replace and CreateOrReplace as well